oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/DefaultOAuth2AuthorizationRequestResolver.java

```diff
@@ -118,6 +119,16 @@ private OAuth2AuthorizationRequest resolve(HttpServletRequest request, String re
 		Map<String, Object> additionalParameters = new HashMap<>();
 		additionalParameters.put(OAuth2ParameterNames.REGISTRATION_ID, clientRegistration.getRegistrationId());
 
+		String queryString = request.getQueryString();
+		String[] params = new String[0];
+		if (!StringUtils.isEmpty(queryString)) {
+			params = queryString.split("&");
+		}
+		for (String param : params){
+			int idx = param.indexOf("=");
+			additionalParameters.put(param.substring(0, idx), param.substring(idx + 1));
```

Comment:
The goal of #5760 is to allow query parameter(s) to be configured in the `authorization-uri` property.

For example:

`authorization-uri: https://accounts.google.com/o/oauth2/v2/auth?access_type=offline`

However, the recent updates in this PR don't fulfill this requirement.
Are you planning on further updates?

Reply:
@mlevkovsky I forgot to mention - the changes I was expecting to see should be in `OAuth2AuthorizationRequest.buildAuthorizationRequestUri()` instead of `DefaultOAuth2AuthorizationRequestResolver`.

@rwinch I agree we should not be parsing URL's and instead reuse `UriComponentsBuilder` for example. The only issue I'm seeing with leveraging `UriComponentsBuilder` is that it lives in `spring-web` which would mean we would need to add that dependency to `oauth2-core`, since `OAuth2AuthorizationRequest` lives there. What are your thoughts on this?

Reply:
I think that is fine to have a dependency on spring-web given OAuth assumes HTTP as a transport. This is especially true of a dependency that is encapsulated.

Reply:
yes sorry for the delays. I'll provide an update as soon as possible
@jgrandja I see your point. I think it's the user's responsibility to provide the proper query params for the appropriate provider.
IMO what the user configures is what we should add to the request uri